### PR TITLE
Flask: add logging to endpoints in routes

### PR DESCRIPTION
**Pull request by delvinso**
part of #361

Initially had most events logged as debug but thought it might be more informative to convert a few to info to ensure things are running smoothly.

### CODE REVIEW

File: flask/app/routes.py [hunk 1]
```diff
@@ -59,9 +65,12 @@ def login():
         user.last_login = datetime.now()
         db.session.commit()
     except:
-        app.logger.warning("Failed to updated last_login for %s", user.username)
+        app.logger.warning("Failed to updated last_login for '%s'", user.username)
```
Comment (Madelinehazel) [inline flask/app/routes.py:68]:
```suggestion
        app.logger.warning("Failed to update last_login for '%s'", user.username)
```
just a small typo

File: flask/app/routes.py [hunk 2]
```diff
@@ -187,48 +203,69 @@ def bulk_update():
...
         except Exception as err:
+            app.logger.error(
+                "CSV failed to be read in. Please verify the file is a properly formatted csv."
+            )
             abort(400, description=str(err))
 
     elif request.content_type == "application/json":
+        app.logger.debug("Content type is JSON..")
         if not request.json:
+            app.logger.debug("Content type is JSON, but request body isn't JSON..")
```
Comment (Madelinehazel) [inline flask/app/routes.py:223]:
```suggestion
            app.logger.error("Content type is JSON, but request body isn't JSON..")
```

File: flask/app/routes.py [hunk 3]
```diff
@@ -187,48 +203,69 @@ def bulk_update():
...
             abort(415, description="Request body must be JSON")
 
         # the jsons must be in a list, even if it is a single json object
         if not isinstance(request.json, list):
+            app.logger.debug(
```
Comment (Madelinehazel) [inline flask/app/routes.py:228]:
```suggestion
            app.logger.error(
```

File: flask/app/routes.py [hunk 4]
```diff
@@ -238,68 +275,120 @@ def bulk_update():
             .all()
         )
         if len(requested_groups) != len(groups):
+            app.logger.debug(
```
Comment (Madelinehazel) [inline flask/app/routes.py:278]:
```suggestion
            app.logger.error(
```

File: flask/app/routes.py [hunk 5]
```diff
@@ -238,68 +275,120 @@ def bulk_update():
...
     else:
+        app.logger.debug("User did not provide groups, checking permissions..")
         groups = (
             models.Group.query.join(models.Group.users)
             .filter(models.User.user_id == user_id)
             .all()
         )
+        app.logger.debug("User is part of '%s'", [x.group_name for x in groups])
         if len(groups) != 1:
+            app.logger.debug(
```
Comment (Madelinehazel) [inline flask/app/routes.py:291]:
```suggestion
            app.logger.error(
```

File: flask/app/routes.py [hunk 6]
```diff
@@ -238,68 +275,120 @@ def bulk_update():
...
+            )
             abort(403, description="User does not belong to any permission groups")
 
+    app.logger.info("Begin processing and inserting records into the database..")
     for i, row in enumerate(dat):
+        app.logger.info("Start Record: %s", i)
+        app.logger.debug("\tChecking sequencing date is provided")
         sequencing_date = row.get("sequencing_date")
         if not sequencing_date:
+            app.logger.debug("\tNo sequencing date provided.")
```
Comment (Madelinehazel) [inline flask/app/routes.py:312]:
```suggestion
            app.logger.error("\tNo sequencing date provided.")
```

**Review by Madelinehazel — COMMENTED** [review]
There's a few instances where I suggested app.logger.error instead of app.logger.debug- feel free to make the changes if you think they make sense

**Comment by delvinso** [comment]
> There's a few instances where I suggested app.logger.error instead of app.logger.debug- feel free to make the changes if you think they make sense

good catch, i agree they should be errors!

**Review by kevinlul — APPROVED** [review]

**Pull request merged**